include/dem/disable_particle_contact.h

```diff
@@ -0,0 +1,239 @@
+    status = mobility_status::mobile;
+    auto cell_iterator_from_mobile_container =
+      status_to_cell[status].find(cell);
+
+    if (cell_iterator_from_mobile_container != status_to_cell[status].end())
+      {
+        return status;
+      }
+
+    // Check if the cell is in the active status set
+    status = mobility_status::active;
+    auto cell_iterator_from_active_container =
+      status_to_cell[status].find(cell);
+
+    if (cell_iterator_from_active_container != status_to_cell[status].end())
+      {
+        return status;
+      }
+
+    // If not mobile or active, cell is inactive
+    return mobility_status::inactive;
+  }
+
+  /**
+   * Convert the vector of mobility status set to a vector of mobility status
+   * with cell id order
+   * status_to_cell is a vector of 3 sets of cell iterator (inactive, active,
+   * mobile), it can't be used as is in the pvd post-processing or any data out,
+   * it needs to be converted to a vector of mobility status by active cell
+   * index
+   *
+   * @param status The initiated vector for the conversion
+   */
+  void
+  get_mobility_status_vector(std::vector<unsigned int> &status)
+  {
+    // Loop over all set with different mobility status (except empty)
+    for (unsigned int i_set = 0; i_set < mobility_status::n_mobility_status - 1;
+         i_set++)
+      {
+        for (auto &cell : status_to_cell[i_set])
+          {
+            if (cell->is_locally_owned())
+              {
+                status[cell->active_cell_index()] = i_set;
+              }
+          }
+      }
+  };
+
+  void
+  get_mobility_status_vector(Vector<float> &status)
+  {
+    // Loop over all set with different mobility status (except empty)
+    for (unsigned int i_set = 0; i_set < mobility_status::n_mobility_status - 1;
+         i_set++)
+      {
+        for (auto &cell : status_to_cell[i_set])
+          {
+            if (cell->is_locally_owned())
+              {
+                status[cell->active_cell_index()] = i_set;
+              }
+          }
+      }
+  };
+
+  std::vector<typename DEM::dem_data_structures<dim>::cell_set>
+  get_mobility_status()
+  {
+    return status_to_cell;
+  }
+
+  LinearAlgebra::distributed::Vector<int>
+  get_mobility_at_nodes()
+  {
+    return mobility_at_nodes;
+  }
+
+  unsigned int
+  get_n_mobility_status()
+  {
+    return mobility_status::n_mobility_status;
+  }
+
+  void
+  set_limit_value(const double granular_temperature,
+                  const double solid_fraction)
+  {
+    granular_temperature_limit = granular_temperature;
+    solid_fraction_limit       = solid_fraction;
+  }
+
+private:
+  void
+  check_granular_temperature(
+    const typename DEM::dem_data_structures<dim>::cells_neighbor_list
+      &                                    cells_neighbor_list,
+    const Particles::ParticleHandler<dim> &particle_handler);
+
+  std::vector<typename DEM::dem_data_structures<dim>::cell_set> status_to_cell;
```

Comment:
I would use cell_mobility_status_map if you decide to use a map here. Otherwise, cell_mobility_status.